### PR TITLE
issue/651-resource-propagates-childs-ignore

### PR DESCRIPTION
**Pull request by FloLey**
# Description

add a new resource entity that raises an IgnoreResourceException when buildings its id

linked to : https://github.com/inmanta/lsm/pull/657

closes https://github.com/inmanta/lsm/issues/651

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [x] Attached issue to pull request
- [x] Changelog entry
- [x] Code is clear and sufficiently documented
- [x] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Correct, in line with design
- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )

### CODE REVIEW

File: pytest_inmanta/module/init.cf [hunk 1]
```diff
@@ -54,3 +54,17 @@ end
 index IgnoreResource(name)
 
 implement IgnoreResource using std::none
+
+
+entity IgnoreResourceInId extends std::PurgeableResource:
+    """
+        This resource will raise an IgnoreResourceException when buildings its id.
```
Comment (Hugo-Inmanta) [inline pytest_inmanta/module/init.cf:61]:
```suggestion
        This resource will raise an IgnoreResourceException when building its id.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,6 @@
 # v 2.9.0 (?)
 Changes in this release:
+- Adds a new resource entity that raises an IgnoreResourceException when buildings its id
```
Comment (Hugo-Inmanta) [inline CHANGELOG.md:3]: I'm not 100% sure whether a changelog entry is relevant here or not

Comment (arnaudsjs) [reply CHANGELOG.md:3]: I would keep it. I agree that it's unlikely that our customers will use this feature, but it's a new feature anyway.

**Review by arnaudsjs — CHANGES_REQUESTED** [review]

**Review by arnaudsjs — CHANGES_REQUESTED** [review]
It looks like you forgot to apply the rename consistency across the other parts of the PR.

File: pytest_inmanta/module/init.py [hunk 3]
```diff
@@ -113,3 +113,19 @@ def get_desired_value(exporter, resource):
...
+    fields = ("name", "desired_value")
+
+    @staticmethod
+    def get_name(exporter, resource):
+        raise resources.IgnoreResourceException()
+
+
+@handler.provider("unittest::IgnoreResourceInIdAttr", name="ignore")
+class IgnoreResourceInIdAttrHandler(handler.CRUDHandler):
+    pass
```
Comment (arnaudsjs) [inline pytest_inmanta/module/init.py:131]: Something you may be unaware of: You can put multiple `@handler.provider` annotations above a single class. So you could just as well have reused the `IgnoreResourceHandler` class for this.

**Review by arnaudsjs — APPROVED** [review]

File: pytest_inmanta/module/init.py (hunk 3 above)
Comment (FloLey) [reply pytest_inmanta/module/init.py]: Indeed, I didn't know. I'll change it

**Pull request merged**